primary/src/block_waiter.rs

```diff
@@ -743,11 +767,12 @@ impl<SynchronizerHandler: Handler + Send + Sync + 'static> BlockWaiter<Synchroni
         let batch_id: BatchDigest = result.clone().map_or_else(|e| e.id, |r| r.id);
 
         match self.tx_pending_batch.remove(&batch_id) {
-            Some((sender, _)) => {
-                debug!("Sending BatchResult with id {}", &batch_id);
-                sender
-                    .send(result)
-                    .expect("Couldn't send BatchResult for pending batch");
+            Some(respond_to) => {
+                for (id, s) in respond_to {
+                    if let Err(err) = s.send(result.clone()) {
+                        error!("Couldn't send batch result {} message to channel [{:?}] for block_id {}", batch_id, err, id);
```

Comment:
Nit: Here and in a few of the lines above you're using a lot of `if let Err(err) = xxx` and logging coupled.

Nothing wrong with that, but some may find it tedious to read. You might want to look at [tap_err](https://docs.rs/tap/latest/tap/tap/trait.TapFallible.html#tymethod.tap_err) and possibly use captured identifiers (`{id}`) to make this a bit lighter. YMMV.

Reply:
In this specific case I am only consuming the error just to print it. Using the `tap_err` in the end is making me (because of clippy) consume the result anyways and do a `let _ = ....` . Maybe the `tap_err` would be much better fit if I wanted to do something further with the result it self. That being said though, I like the proposal of introducing [tap](https://crates.io/crates/tap) as I can definitely see cases where we can use it - thanks for the recommendation! I've applied the recommendation.